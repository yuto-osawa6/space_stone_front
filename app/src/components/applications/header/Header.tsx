import { useEffect, useRef, useState } from "react";
import { BsFillSuitHeartFill, BsListUl } from "react-icons/bs"
import { MdDriveFileMoveOutline, MdHome, MdOutlineArticle, MdRateReview, MdSearch } from "react-icons/md";
import { useDispatch } from "react-redux";
import { useScrollPosition } from '@n8tb1t/use-scroll-position';
import { HiOutlineSearchCircle } from "react-icons/hi";
import { IoMdMenu } from "react-icons/io";
import Link from 'next/link'
import { AiOutlineComment, AiOutlineHome } from "react-icons/ai"
import { RiArticleLine } from "react-icons/ri"
import { UserCertification } from "../user/UserCertification";
import { SubMenuAction } from "@/store/submenu/actions";
import { useLocale } from "@/lib/ini/local/local";
import { useWindowDimensions } from "@/hook/useWindowResize";
import { url } from "@/utils/config";
import { useScroll } from "@/hook/useScroll";
import { useRouter } from "next/router";
import { useExecLeft } from "@/lib/api/left";
import LeftStyle from "../left/leftMenus/LeftStyles";
import LeftGenre from "../left/leftMenus/LeftGenres";
import { LeftsArticles } from "@/components/applications/left/leftMenus/LeftArticles";
import { LeftsReviews } from "../left/leftMenus/LeftReviews";
import { LeftsThreads } from "../left/leftMenus/LeftsThreads";


type Props = {
  locationNumber: number | undefined
}
export const Header:React.FC<Props> = function HeaderFunc(Props){
  const a = useScroll()
  // console.log(a)
  // useScrollPosition
  const dispatch = useDispatch()
  const [showMenu, setShowMenu] = useState<boolean>(true);
  const [currentPostion,setCurrentPositon] = useState<number>(0)
  // useScroll(({scrollY,preveScroll}) => {
  //   // if (scrollY>-1){
  //   //   setShowMenu(true)
  //   //   return
  //   // }
  //   // const visible = scrollY > preveScroll;
  //   // setShowMenu(visible);
  // },[])

  useEffect(()=>{
    // console.log("aaa")
    // console.log(a.scrollY ,a.prevScrollY )
    // console.log(showMenu)
    // console.log(a.scrollY<1)
    // console.log("aaa")
    // console.log(a.scrollY > a.prevScrollY)
    if (a.scrollY<56){
      console.log("aaa34")
      setShowMenu(true)
      return
    }
    const visible = a.scrollY < a.prevScrollY
    setShowMenu(visible);
  },[a.scrollY])

  useScrollPosition(({ prevPos, currPos }) => {
    setCurrentPositon(currPos.y)
    if (currPos.y>-1){
      // setShowMenu(true)
      return
    }
    const visible = currPos.y > prevPos.y;
    // setShowMenu(visible);
  }, []);

  // submenu
  const submenuref = useRef<HTMLLIElement>(null)
  const [openMenu,setOpenMenu] = useState<boolean>(false)
  useEffect(() => {
    const checkIfClickedOutside = (e:any) => {
      if (openMenu && submenuref.current && !submenuref.current.contains(e.target)) {
        setOpenMenu(false);
        // document.body.style.overflowY = ""
        // document.body.style.touchAction = ""
      }
    };
    document.addEventListener("mousedown", checkIfClickedOutside);
    return () => {
      document.removeEventListener("mousedown", checkIfClickedOutside);
    };
  }, [openMenu]); 

  const windowSize = useWindowDimensions()
  const [width,setWidth] = useState<boolean>()
  function handle(event:any) {
    event.preventDefault();
}
  const setOpenMenuHandler = (e:React.MouseEvent<HTMLLIElement> | undefined) => {
    e?.stopPropagation()
    // console.log("aaaae2")
    openMenu==true?setOpenMenu(false):setOpenMenu(true)
    if (windowSize.width < 768){
      openMenu==false?document.body.style.overflow = "hidden":document.body.style.overflow = ""
      openMenu==false?document.body.style.height = "100%":document.body.style.overflow = ""
      // openMenu==false?document.body.style.touchAction = "none":document.body.style.touchAction = ""
    }else{
    }
  }

  const w = useWindowDimensions()
  const handleStyle = () => {
    if(Props.locationNumber===undefined){
    return {}
    }else if(Props.locationNumber===1){
    return {display:"none"}
    // return {backgroundColor:"#1a252f"}
    }
  }
  const handleStyle2 = () => {
    if(Props.locationNumber===undefined){
    return {}
    }else if(Props.locationNumber===1){
      if(w.width > 768&&currentPostion>-1){
        return {width:"100%",backgroundColor: "rgb(246 246 249 / 30%)"}
      }else if(w.width > 768){
        return {width:"100%",backgroundColor:"#f6f6f9"}
      }else{
        return {width:"100%",backgroundColor: "transparent"}
      }
    }
}
const router = useRouter()
const handleClick = (url:string) => {
  router.push(url)
  dispatch(SubMenuAction(true))
}
// locate
const { t } = useLocale()
const routerHome = () => {
  router.push("/")
}
// console.log(windowSize)
    // useEffect(()=>{
    //   const touchHandler = (event: any) => {
    //     // if(windowSize.width >= 768){
    //     //   return
    //     // }else{
    //       event.preventDefault();
    //     // }
    //   };
    //   document.addEventListener('touchmove', touchHandler, {
    //     passive: false
    //   });
    //   return () => {
    //     document.removeEventListener("touchmove", touchHandler);
    //   };
    // },[windowSize.width])
    const {data,error} = useExecLeft()
    console.log(data)
  return(
    <>
      <div className = "HeaderV1">
        <div className = "HeaderMain"
          style={handleStyle()}
        >
          <div className = "HeaderMainLeft">
            <div className = "HeaderMainLeftTitle">
              <div className = "LOGO">
                <div className=""
                style={{
                  marginRight:"10px"
                }}
                >
                <img src={url('/topimage.png')} alt="Sample image"></img>
                </div>
                <div className = "LogoG">
                  {t.Logo.G}
                </div>
                <div className = "LogoF">
                  {t.Logo.F}
                </div>
                {/* <div className = "LogoHeart">
                  <BsFillSuitHeartFill/>
                </div> */}
              </div>
            </div>
          </div>
        </div>
        <div className = {`HeaderNavi`}  
        style={{position:"relative"}}
        >
          {Props.locationNumber!=1&&(
          <div className = "HeaderMainLeftTitle">
              <div className = "LOGO">
                <div className=""
                onClick={routerHome}
                style={{
                  marginRight:"10px"
                }}
                >
                <img src={url('/topimage.png')} alt="Sample image"></img>
                </div>
                {/* <div className = "LogoG">
                  {t.Logo.G}
                </div>
                <div className = "LogoF">
                  {t.Logo.F}
                </div> */}
                {/* <div className = "LogoHeart">
                  <BsFillSuitHeartFill/>
                </div> */}
            </div>
          </div>
          )}
          <ul className={showMenu?"":"activeScroll"} 
          style={
            handleStyle2()
          }
          >
            <li className = "nomalNavi"><Link href="/"><a><div><div><AiOutlineHome/> {t.Headers.TOP}</div><div className = {"home"}>{t.SubHeader.TOP}</div></div></a></Link></li>
            <li className = "nomalNavi"><Link href="/search"><a><div><div><HiOutlineSearchCircle/> {t.Headers.SEARCH}</div><div className = {"home"}>{t.SubHeader.SEARCH}</div></div></a></Link></li>
            <li className = "nomalNavi"><Link href="/articles"><a><div><div><RiArticleLine/> {t.Headers.ARTICLE}</div><div className = {"home"}>{t.SubHeader.ARTICLE}</div></div></a></Link></li>
            <li className = "nomalNavi"><Link href="/reviews"><a><div><div><AiOutlineComment/> {t.Headers.REVIEWS}</div><div className = {"home"}>{t.SubHeader.REVIEWS}</div></div></a></Link></li>
            <li className = "nomalNavi"><Link href="/threads"><a><div><div><AiOutlineComment/> {t.Headers.THREAD}</div><div className = {"home"}>{t.SubHeader.THREAD}</div></div></a></Link></li>
            
            <li className="headerUserSighIn nomalNavi"
            style={{
              display:"block"
            }}
            ><UserCertification/></li>
            <li 
              className = "subMenu"
              ref={submenuref}
              onClick={setOpenMenuHandler}
            > 
              <IoMdMenu    
              />
              {windowSize.width >= 768&&openMenu==true&&(
              <div className = "subMenuList"
              >
                <div className="">
                  <div onClick={()=>handleClick("/#weekly-ranking")}><a>{t.SubMenu.QUESTIONNAIRE}</a></div>
                  <div onClick={()=>handleClick("/#this-season")}><a>{t.SubMenu.THISSEASON}</a></div>
                  <div onClick={()=>handleClick("/#last-season")}><a>{t.SubMenu.LASTSEASON}</a></div>
                  <div onClick={()=>handleClick("/#next-season")}><a>{t.SubMenu.NEXTSEASON}</a></div>
                  <div onClick={()=>handleClick("/#movies")}><a>{t.SubMenu.MOVIE}</a></div>
                  <div onClick={()=>handleClick("/#news")}><a>{t.SubMenu.NEWS}</a></div>
                  <div onClick={()=>handleClick("/#toptens")}><a>{t.SubMenu.TOP10}</a></div>
                  <div className = "Top100SubMenu"><Link href ="/top100"><a>{t.SubMenu.TOP100}</a></Link></div>
                  <div className = "Top100SubMenu"><Link href ="/tier"><a>{t.SubMenu.TIER}</a></Link></div>
                  <div className = "Top100SubMenu"><Link href ="/weekly"><a>{t.SubMenu.LASTQUESTIONNAIRE}</a></Link></div>
                </div>
              </div>
              )}

                {windowSize.width < 768&&openMenu==true&&(   
                  <>
                  <div className = "subMenuList768"
                  >
                    <div className="subMenuList7681">Main Menu</div>
                  
                    <div className="">
                      <div className = "subMenuList7689"><Link href="/"><a><div className="headersub768"><div className = "home2"><AiOutlineHome/> {t.Headers.TOP}</div><div className = {"home23"}>{t.SubHeader.TOP}</div></div></a></Link>
                        {/* <div className="">Home Menu</div> */}
                        <div className="HederV1SubmenuSub">
                          <div onClick={()=>handleClick("/#weekly-ranking")}><a>{t.SubMenu.QUESTIONNAIRE}</a></div>
                          <div onClick={()=>handleClick("/#this-season")}><a>{t.SubMenu.THISSEASON}</a></div>
                          <div onClick={()=>handleClick("/#last-season")}><a>{t.SubMenu.LASTSEASON}</a></div>
                          <div onClick={()=>handleClick("/#next-season")}><a>{t.SubMenu.NEXTSEASON}</a></div>
                          <div onClick={()=>handleClick("/#movies")}><a>{t.SubMenu.MOVIE}</a></div>
                          <div onClick={()=>handleClick("/#news")}><a>{t.SubMenu.NEWS}</a></div>
                          <div onClick={()=>handleClick("/#toptens")}><a>{t.SubMenu.TOP10}</a></div>
                        </div>
                      </div>
                      <div className = "subMenuList7689"><Link href="/search"><a><div className="headersub768"><div className = "home2"><HiOutlineSearchCircle/> {t.Headers.SEARCH}</div><div className = {"home23"}>{t.SubHeader.SEARCH}</div></div></a></Link>
                        <div className="HederV1SubmenuSub">
                          {data.styles.map((item: any) => (
                            <LeftStyle name={item.name} id={item.id} count={item.count} key={item.id}/>
                          ))}
                        </div>
                      </div>
                      <div className = "subMenuList7689"><Link href="/articles"><a><div className="headersub768"><div className = "home2"><RiArticleLine/> {t.Headers.ARTICLE}</div><div className = {"home23"}>{t.SubHeader.ARTICLE}</div></div></a></Link>
                        <div className="HederV1SubmenuSub">
                          <LeftsArticles/>
                        </div>
                      </div>
                      <div className = "subMenuList7689"><Link href="/reviews"><a><div className="headersub768"><div className = "home2"><AiOutlineComment/> {t.Headers.REVIEWS}</div><div className = {"home23"}>{t.SubHeader.REVIEWS}</div></div></a></Link>
                        <div className="HederV1SubmenuSub">
                          <LeftsReviews/>
                        </div>
                      </div>
                      <div className = "subMenuList7689"><Link href="/threads"><a><div className="headersub768"><div className = "home2"><AiOutlineComment/> {t.Headers.THREAD}</div><div className = {"home23"}>{t.SubHeader.THREAD}</div></div></a></Link>
                        <div className="HederV1SubmenuSub">
                          <LeftsThreads/>
                        </div>
                      </div>
                      <div className = "Top100SubMenu subMenuList7689"><Link href ="/top100"><a><div className="headersub768"><div className = "home2"><AiOutlineHome/>{t.SubMenu.TOP100}</div><div className = {"home23"}></div></div></a></Link></div>
                      <div className = "Top100SubMenu subMenuList7689"><Link href ="/tier"><a><div className="headersub768"><div className = "home2"><AiOutlineHome/>{t.SubMenu.TIER}</div><div className = {"home23"}></div></div></a></Link></div>
                      <div className = "Top100SubMenu subMenuList7689"><Link href ="/weekly"><a><div className="headersub768"><div className = "home2"><AiOutlineComment/> {t.SubMenu.QUESTIONNAIRE}</div><div className = {"home23"}></div></div></a></Link></div>
                    </div>
                  </div>
                  
                  </>  
                )}
            </li>
          </ul>
        </div>
      </div>
    </>
  )
}