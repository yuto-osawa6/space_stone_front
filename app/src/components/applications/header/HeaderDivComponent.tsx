import { ReactNode } from "react"

type Props = {
  children:ReactNode
}

export const HeaderDivComponent:React.FC<Props> = (Props) => {

  return(
    <>
    <div className = "header">
      <div className = "header__box">
        {/* <Header/> */}
        {Props.children}
      </div>
    </div>
    </>
  )
}