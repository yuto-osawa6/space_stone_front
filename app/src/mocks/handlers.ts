import { rest } from "msw";
import { mockProductReds } from "@/mocks/api/product_red"
import { execGoogleMock, getCurrentUserMock, signOutMock } from "./api/user/signin";
import { execLeftMock } from "./api/left/left_mock";
// import sucessResponse from "./sucessResponse.json";

export const API = "http://localhost:3001/api/v1"
export const APISOCIAL = "http://localhost:3001"
export const API2 = "http://api:3001/api/v1"

export const handlers = [
  rest.get(`${API}/products/red`, mockProductReds),
  rest.get(`${API2}/products/red`, mockProductReds),
  rest.post(`${APISOCIAL}/social_auth/callback`,execGoogleMock),
  rest.get(`${API}/session_user`,getCurrentUserMock),
  rest.delete(`${APISOCIAL}/auth/sign_out`,signOutMock),
  rest.get(`${API}/products/left`,execLeftMock),

  // rest.get("https://sample/api/list", (_, res, ctx) => {
  //     return res(
  //       ctx.json(sucessResponse),
  //     );
  //   },
  // ),
];