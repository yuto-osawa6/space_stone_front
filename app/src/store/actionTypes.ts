// *
// * action types
// * 一意となるキーを指定するので、Actionが増えるたびにここにキーを書く
// *

export const ActionTypes = {
  increment: "INCREMENT", // "INCREMENT"型
  decrement: "DECREMENT", // "DECREMENT"型
  countReset: "COUNT_RESET", // "COUNT_RESET"型

  searchProductsTexts: "SEARCH_PRODUCTS_TEXT",
  // searchProductsStyles: "SEARCH_PRODUCTS_STYLE",
  // pussingGenresData:"PUSSING_GENRES_DATA",
  searchClearTitle:"SEARCH_CLEAR_TITLE",

  setGenres: "SET_GENRES",
  setFFF:"Set_FFF",

  setGenre: "SET_GENRES2",
  setCCC:"Set_FFF2",

  todoGenres:"TODO_GENRES",
  deletetodoGenres:"DELETE_TODO_GENRES",
  deletetodoGenresOne:"DELETE_TODO_GENRES_ONE",
  deletetodoGenresAll:"DELETE_TODO_GENRES_ALL",
  deletetodoGenresExceptOne:"DELETE_TODO_GENRES_Except_One",




  setStyles: "SETSTYLES",

  todoStyles:"TODO_STYLES",
  deletetodoStyles:"DELETE_TODO_STYLES",
  deletetodoStylesAll:"DELETE_TODO_STYLES_ALL",

  sort : "SORT",

  grid :"GRID",

  user : "User",
  updateNickName : "UPDATE_NICK_NAME",
  updateBackgriundImage : "UPDATE_BACKGROUND_IMAGE",
  updateOverview : "UPDATE_OVER_VIEW",


  product:"SETPRODUCT",
  productshow: "SETPRODUCT2",

  Casts:"CASTS",
  DeleteCasts:"DELETE_CASTS",
  DeleteCastsOne:"DELETE_CASTS_ONE",
  DeleteCastsAll:"DELETE_CASTS_ALL",
  DeleteCastsExceptOne:"DELETE_CASTS_EXCEPT_ONE",

  DeliveryEndGteq:"DELIVERY_END_GTEQ",
  DeliveryStartGteq:"DELIVERY_START_GTEQ",
  PickupSearch:"PICKUP_SEARCH",
  PrivateSearch:"PRIVATE_SEARCH",
  NewSearch:"NEW_SEARCH",

  YearSearch:"YEAR_SEARCH",
  YearSearchDestroy:"YEAR_SEARCH_DESTROY",
  DuringSearch:"DURING_SEARCH",
  DuringSearchDestroy:"DURING_SEARCH_DESTROY",
  SeasonSearch:"SEASON_SEARCH",
  SeasonSearchDestroy:"SEASON_SEARCH_DESTROY",
  AllDeleteSubClass:"ALL_DELETE_SUB_CLASS",

  selectedArticle:"SELECTED_ARTICLE",
  deleteArticle:"DELETE_ARTICLE",

  // lefts navigate
  NavigateLeftArticle:"NAVIGATE_LEFT_ARTICLE",
  DeleteLeftArticle:"DELETE_LEFT_ARTICLE",

  NavigateLeftReviewData:"NAVIGATE_LEFT_REVIEW",
  DeleteLeftReview:"DELETE_LEFT_REVIEW",

  NavigateLeftThreadData:"NAVIGATE_LEFT_THREAD",
  DeleteLeftThread:"DELETE_LEFT_THREAD",

  // Manage subsearch
  OpenSubSearch:"OpenSubSearch",

  // sortPeriod
  setSortPeriod:"SET_SORT_PERIOD",
  resetSortPeriod:"RESET_SORT_PERIOD",

  // studio

  Studios:"STUDIOS",
  DeleteStudios:"DELETE_STUDIOS",
  DeleteStudiosOne:"DELETE_STUDIOS_ONE",
  DeleteStudiosAll:"DELETE_STUDIOS_ALL",
  DeleteStudiosExceptOne:"DELETE_STUDIOS_EXCEPT_ONE",


  // kisetsu
  todoKisetsu:"TODO_KISETSU",
  // deletetodo:"DELETE_TODO_STYLES",
  deletetodoKisetsuAll:"DELETE_TODO_KISETSU_ALL",
  

  // emotionSort
  emotionSort:"EMOTION_SORT",
  deleteEmotionSort:"DELETE_EMOTION_SORT",

  // reviewUpdate
  updateReviewState:"UPDATE_REVIEW_STATE",
  updateThreadState:"UPDATE_THREAD_STATE",

  // message
  pusshingMessage:"PUSSHING_MESSAGE",
  deleteingMessage:"DELETING_MESSAGE",
  selectDeleteMessage:"SELECT_DELETE_MESSAGE"

} as const;