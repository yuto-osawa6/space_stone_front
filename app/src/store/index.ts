import { combineReducers, createStore } from 'redux';

import { countReducer } from './counter/reducer';
import { searchProdcutsReducer } from './search/reducer';
// import { SettingGenresDatetypes } from './genres/types';
import { settiongGenresReducer } from './genres/reducer';
import { settiongtodoGenresReducer } from './todogenres/reducer';
import { settiongStylesReducer } from './styles/reducer';
import { settiongtodoStylesReducer } from './todostyles/reducer';
// import { sortEmotionReducer, sortReducer } from './sort/reducer';
import { GridReducer } from './grid/reducer';
import { userReducer } from './user/reducer';
import { settiongProductReducer } from './product/reducer';
import { SettingCastsReducer } from './casts/reducer';
import { SettiongSubSearchesReducer } from './subsearches/reducer';
import { YearSearchReducer } from './year/reducer';
import { SeasonSearchReducer } from './season/reducer';
import { TimeSearchReducer } from './during/reducer';
import { SettingArticleReducer } from './article/reducer';
import { SettingLeftArticleReducer } from './lefts/article/reducer';
import { SettingLeftReviewReducer } from './lefts/review/reducer';
import { SettingLeftThreadReducer } from './lefts/thread/reducer';
import { SettiongOpenCloseManegementSubSearchReducer } from './subsearchopenmanagement/reducer';
import { SortPeriodReducer } from './sortperiod/reducer';
import { SettingStudiosReducer } from './studios/reducer';
import {settiongtodoKisetsuReducer} from './kisetsu/reducer'
import { sortEmotionReducer } from './emotion/reducer';
import { sortReducer } from './sort/reducer';
import { updateReviewReducer } from './reviewUpdate/reducer';
import { updateThreadReducer } from './updateThread/reducer';
import { topMessageReducer } from './message/reducer';
import { subMenuReducer } from './submenu/reducer';
// import { settiongtodoGenresReducer } from './todogenres/reducer';
// import { settiongtododGenresReducer } from './todogenre/reducer';



// import { settiongGenresReducer2 } from './genre/reducer';


// *
// * store 本体
// *

// Reducerを増やすときは、ここに追加
const rootReducer = combineReducers({
  counter: countReducer,
  search : searchProdcutsReducer,
  genres : settiongGenresReducer,
  styles : settiongStylesReducer,
  todogenres:settiongtodoGenresReducer,
  todostyles:settiongtodoStylesReducer,
  sort : sortReducer,
  grid : GridReducer,
  user :userReducer,

  product :settiongProductReducer,

  cast : SettingCastsReducer,
  subsearch : SettiongSubSearchesReducer,
  // todogenre:settiongtododGenresReducer,
  yearsearch : YearSearchReducer,
  timesearch :  TimeSearchReducer,
  seasonsearch :SeasonSearchReducer,
  article : SettingArticleReducer,
  leftArticle : SettingLeftArticleReducer,
  leftReview:SettingLeftReviewReducer,
  leftThread:SettingLeftThreadReducer,
  OpenCloseSubSearch:SettiongOpenCloseManegementSubSearchReducer,
  SortPeriod:SortPeriodReducer,
  // genre : settiongGenresReducer2
  studios:SettingStudiosReducer,
  kisetsu:settiongtodoKisetsuReducer,
  sortEmotion:sortEmotionReducer,
  updateReview:updateReviewReducer,
  updateThread:updateThreadReducer,
  message:topMessageReducer,
  submenu:subMenuReducer

});

// states type
export type RootState = ReturnType<typeof rootReducer>; // ReturnType<typeof fn>は、fnの返り値の型

// store
const store = createStore(rootReducer);

export default store;