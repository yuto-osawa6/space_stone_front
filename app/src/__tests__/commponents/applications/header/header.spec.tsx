import { Header } from "@/components/applications/header/Header";
import { server } from "@/mocks/server";
import store from "@/store";
import {render, fireEvent, waitFor, screen, cleanup} from '@testing-library/react'
import { Provider } from "react-redux";
import renderer from 'react-test-renderer';
import { getPage } from "next-page-tester";
import "whatwg-fetch"

// import * as nextRouter from 'next/router';

// nextRouter.useRouter = jest.fn();
// nextRouter.useRouter.mockImplementation(() => ({ route: '/' }));
jest.mock("next/router", () => ({
  useRouter() {
      return {
          route: "/",
          pathname: "",
          query: "",
          asPath: "",
          locale:"ja"
      };
  },
}));

// import "whatwg-fetch"
// import { getPage } from 'next-page-tester';
// import { initTestHelpers } from 'next-page-tester'; 
// initTestHelpers();
describe('Header', () => {
  beforeAll(() => server.listen())
  afterEach(() => {
    server.resetHandlers()
    cleanup()
  })
  afterAll(() => server.close())

  test('snapshot', async() => {
    const { container } = render(<Provider store={store}>
      <Header
      locationNumber={1}
      />
    </Provider>)
    screen.debug();
    // await waitFor(() => screen.findByText(/Signed in as/))
    expect(container).toMatchSnapshot()

   
  });
  // test('snapshot', async() => {
  //   const { container } = render(<Ota2/>)
  //   // await waitFor(() => screen.findByText(/Signed in as/))
  //   expect(container).toMatchSnapshot()
  // });
});