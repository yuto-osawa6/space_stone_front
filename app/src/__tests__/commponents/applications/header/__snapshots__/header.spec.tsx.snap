// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<div>
  <div
    class="HeaderV1"
  >
    <div
      class="HeaderMain"
      style="display: none;"
    >
      <div
        class="HeaderMainLeft"
      >
        <div
          class="HeaderMainLeftTitle"
        >
          <div
            class="LOGO"
          >
            <div
              class=""
              style="margin-right: 10px;"
            >
              <img
                alt="Sample image"
                src="undefined/topimage.png"
              />
            </div>
            <div
              class="LogoG"
            >
              Meru
            </div>
            <div
              class="LogoF"
            >
              Planet
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="HeaderNavi"
      style="position: relative;"
    >
      <ul
        class=""
        style="width: 100%; background-color: transparent;"
      >
        <li
          class="nomalNavi"
        >
          <a
            href="/"
          >
            <div>
              <div>
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                  />
                </svg>
                 
                Home
              </div>
              <div
                class="home"
              >
                ホーム
              </div>
            </div>
          </a>
        </li>
        <li
          class="nomalNavi"
        >
          <a
            href="/search"
          >
            <div>
              <div>
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 16l2.879-2.879m0 0a3 3 0 104.243-4.242 3 3 0 00-4.243 4.242zM21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                 
                Search
              </div>
              <div
                class="home"
              >
                検索
              </div>
            </div>
          </a>
        </li>
        <li
          class="nomalNavi"
        >
          <a
            href="/articles"
          >
            <div>
              <div>
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M20 22H4a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1h16a1 1 0 0 1 1 1v18a1 1 0 0 1-1 1zm-1-2V4H5v16h14zM7 6h4v4H7V6zm0 6h10v2H7v-2zm0 4h10v2H7v-2zm6-9h4v2h-4V7z"
                    />
                  </g>
                </svg>
                 
                Article
              </div>
              <div
                class="home"
              >
                記事
              </div>
            </div>
          </a>
        </li>
        <li
          class="nomalNavi"
        >
          <a
            href="/reviews"
          >
            <div>
              <div>
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  t="1569682881658"
                  version="1.1"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs />
                  <path
                    d="M573 421c-23.1 0-41 17.9-41 40s17.9 40 41 40c21.1 0 39-17.9 39-40s-17.9-40-39-40zM293 421c-23.1 0-41 17.9-41 40s17.9 40 41 40c21.1 0 39-17.9 39-40s-17.9-40-39-40z"
                  />
                  <path
                    d="M894 345c-48.1-66-115.3-110.1-189-130v0.1c-17.1-19-36.4-36.5-58-52.1-163.7-119-393.5-82.7-513 81-96.3 133-92.2 311.9 6 439l0.8 132.6c0 3.2 0.5 6.4 1.5 9.4 5.3 16.9 23.3 26.2 40.1 20.9L309 806c33.5 11.9 68.1 18.7 102.5 20.6l-0.5 0.4c89.1 64.9 205.9 84.4 313 49l127.1 41.4c3.2 1 6.5 1.6 9.9 1.6 17.7 0 32-14.3 32-32V753c88.1-119.6 90.4-284.9 1-408zM323 735l-12-5-99 31-1-104-8-9c-84.6-103.2-90.2-251.9-11-361 96.4-132.2 281.2-161.4 413-66 132.2 96.1 161.5 280.6 66 412-80.1 109.9-223.5 150.5-348 102z m505-17l-8 10 1 104-98-33-12 5c-56 20.8-115.7 22.5-171 7l-0.2-0.1C613.7 788.2 680.7 742.2 729 676c76.4-105.3 88.8-237.6 44.4-350.4l0.6 0.4c23 16.5 44.1 37.1 62 62 72.6 99.6 68.5 235.2-8 330z"
                  />
                  <path
                    d="M433 421c-23.1 0-41 17.9-41 40s17.9 40 41 40c21.1 0 39-17.9 39-40s-17.9-40-39-40z"
                  />
                </svg>
                 
                Review
              </div>
              <div
                class="home"
              >
                レビュー
              </div>
            </div>
          </a>
        </li>
        <li
          class="nomalNavi"
        >
          <a
            href="/threads"
          >
            <div>
              <div>
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  t="1569682881658"
                  version="1.1"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs />
                  <path
                    d="M573 421c-23.1 0-41 17.9-41 40s17.9 40 41 40c21.1 0 39-17.9 39-40s-17.9-40-39-40zM293 421c-23.1 0-41 17.9-41 40s17.9 40 41 40c21.1 0 39-17.9 39-40s-17.9-40-39-40z"
                  />
                  <path
                    d="M894 345c-48.1-66-115.3-110.1-189-130v0.1c-17.1-19-36.4-36.5-58-52.1-163.7-119-393.5-82.7-513 81-96.3 133-92.2 311.9 6 439l0.8 132.6c0 3.2 0.5 6.4 1.5 9.4 5.3 16.9 23.3 26.2 40.1 20.9L309 806c33.5 11.9 68.1 18.7 102.5 20.6l-0.5 0.4c89.1 64.9 205.9 84.4 313 49l127.1 41.4c3.2 1 6.5 1.6 9.9 1.6 17.7 0 32-14.3 32-32V753c88.1-119.6 90.4-284.9 1-408zM323 735l-12-5-99 31-1-104-8-9c-84.6-103.2-90.2-251.9-11-361 96.4-132.2 281.2-161.4 413-66 132.2 96.1 161.5 280.6 66 412-80.1 109.9-223.5 150.5-348 102z m505-17l-8 10 1 104-98-33-12 5c-56 20.8-115.7 22.5-171 7l-0.2-0.1C613.7 788.2 680.7 742.2 729 676c76.4-105.3 88.8-237.6 44.4-350.4l0.6 0.4c23 16.5 44.1 37.1 62 62 72.6 99.6 68.5 235.2-8 330z"
                  />
                  <path
                    d="M433 421c-23.1 0-41 17.9-41 40s17.9 40 41 40c21.1 0 39-17.9 39-40s-17.9-40-39-40z"
                  />
                </svg>
                 
                Thread
              </div>
              <div
                class="home"
              >
                スレッド
              </div>
            </div>
          </a>
        </li>
        <li
          class="headerUserSighIn nomalNavi"
          style="display: block;"
        />
        <li
          class="subMenu"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M64 384h384v-42.666H64V384zm0-106.666h384v-42.667H64v42.667zM64 128v42.665h384V128H64z"
            />
          </svg>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
